cpp/include/Ice/Timer.h

```diff
@@ -43,6 +55,13 @@ namespace Ice
         // Destroy the timer and join the timer execution thread. Must not be called from a timer task.
         void destroy();
 
+        // Schedule a function-task for execution after a given delay.
+        template<class Rep, class Period>
+        void schedule(std::function<void()> function, const std::chrono::duration<Rep, Period>& delay)
```

Comment:
In the original DataStorm Timer API, schedule returned a "cancel" function. This supports cancellation while storing std::function. 

We could refactor IceInternal::Timer to use this old DataStorm API but it would be much more work. And it would be difficult to implement isScheduled.